File: cpp/base/src/vsdk/toolkit/tonemap/IdentityToneMap.cpp
#include "vsdk/toolkit/tonemap/IdentityToneMap.h"

IdentityToneMap::IdentityToneMap() {
}

IdentityToneMap::~IdentityToneMap() {
}

void
IdentityToneMap::init(const ToneMappingContext &/*toneMapOptions*/) {
}

ColorRgbMutable
IdentityToneMap::scaleForComputations(ColorRgbMutable radiance) const {
    return radiance;
}

ColorRgbMutable
IdentityToneMap::scaleForDisplay(ColorRgbMutable radiance) const {
    return radiance;
}

File: typescript/base/src/main/vsdk/toolkit/tonemap/IdentityToneMap.ts
import { ColorRgb } from "../common/color/ColorRgb";
import { ToneMap } from "./ToneMap";
import { ToneMappingContext } from "./ToneMappingContext";

export class IdentityToneMap extends ToneMap {
  public constructor() {
    super();
  }

  public override init(toneMapOptions: ToneMappingContext): void {
    void toneMapOptions;
  }

  public override scaleForComputations(radiance: ColorRgb): ColorRgb {
    return radiance;
  }

  public override scaleForDisplay(radiance: ColorRgb): ColorRgb {
    return radiance;
  }
}
